ddtrace/tracer/payload.go

```diff
@@ -53,8 +53,26 @@ func newPayload() *payload {
 	return p
 }
 
+// populateTraceTags puts trace tags in the first span until we adapt the new payload format.
+func populateTraceTags(t spanList) {
```

Comment:
These tags should be populated at the time when they are set, not as a "payload fix" at the end.